cmd/namespace-lock.go

```diff
@@ -56,39 +60,50 @@ type RWLockerSync interface {
 // Returns lock clients and the node index for the current server.
 func newDsyncNodes(endpoints EndpointList) (clnts []dsync.NetLocker, myNode int) {
 	cred := globalServerConfig.GetCredential()
-	clnts = make([]dsync.NetLocker, len(endpoints))
 	myNode = -1
-	for index, endpoint := range endpoints {
+	uniqueClnts := set.NewStringSet()
+	uniqueServers := set.NewStringSet()
```

Comment:
It is not needed to use two sets for this code. Just one `seenHosts` is enough to collect the hosts the first time they are seen. Whether the host turns out to be a server or a client, it should be processed only once in the following loop.

Reply:
how would you do it? @donatello with only once..